string2.py

```diff
@@ -43,42 +50,54 @@ def not_bad(s):
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
-    """Your code goes here.  Edit this docstring."""
-    return
-
-
-# Provided simple test() function used in main() to print
+  if len(a)%2 == 0:
```

Comment:
There is a lot of `len(a)/2` and len(b)/2 computing going on -- re-computing the same thing.  You could store these results into a variable so you don't have to keep doing this computation ...